### DISCUSSION TITLE
Javascript support [feature request]

### DISCUSSION THREAD

**Discussion by solmoller** [inseri-swiss/inseri-core-wp#198] [Ideas]
As much of the usage of the package will be in a webbrowser, a javascript component could be very handy.

Python is sluggish for data handling, javascript is much faster when trading directly in a browser.

BR

Henrik

**Comment by aplamada** [comment]
A JavaScript Code block idea might be an interesting one. In which context do you want to use it and for which concrete use case?

**Reply by solmoller** [reply]
Javascript is fairly often the default scripting tool in web-development, and compared to python the code is much faster. 

My specific use case is to fetch some json data from different sources. As the data is of different origin and quality, aligning and combining data is a cumbersome task, but it's done and working in a browser now. 

Next step is plotly in Wordpress, and hey presto, now I am reading how you managed to pull off python in Wordpress. Impressively compact code you have written!


More specific:  I am publishing a daily power price prognosis, and to combine that with taxes and specific transportation costs from 30+ utility companies is painful. I am a nonprofit trying to push Danes into shifting their power consumption to the cheapest days with lowest CO2/kWh.

BR

Henrik

**Reply by aplamada** [reply]
@solmoller Thanks for the description and I happy to hear that you find your way.

[JavaScript Code Block](https://github.com/inseri-swiss/inseri-swiss/issues/122) is part of the roadmap. I hope to update you in the next month.

**Reply by aplamada** [reply]
@solmoller Please see [v0.3.5](https://github.com/inseri-swiss/inseri-core-wp/releases) where we added JavaScript Code block. It works similar to the Python Code block. To import a package please use cdn.jsdelivr.net. For example to import Day.js:
```js
importScripts('https://cdn.jsdelivr.net/npm/dayjs@1.11.10/dayjs.min.js')
console.log(dayjs().format())
```

Let us know your feedback!
